Add new configuration settings and update the default of `server.metrics.filter` (#2180)
neo4j/docs-operations: modules/ROOT/pages/configuration/configuration-settings.adoc

m|+++600+++
|===

== Cloud storage integration settings

> Configuration settings enable custom cloud storage and host authority endpoints.

Author: Cloud integration settings allow you to specify custom Azure blob storage endpoints and host authorities, set the project ID for Google Cloud Storage buckets, and define the desired throughput for transfer operations in Amazon S3.

> Configuration settings enable custom cloud storage and host authority endpoints.

Author: ```suggestion
Cloud integration settings allow you to specify custom Azure blob storage endpoints and host authorities, set the project ID for Google Cloud Storage buckets, and define the desired throughput for transfer operations in Amazon S3.
```

[role=label--enterprise-edition label--new-2025.03]
[[config_dbms.integrations.cloud_storage.azb.blob_endpoint_suffix]]
=== `dbms.integrations.cloud_storage.azb.blob_endpoint_suffix`

.dbms.integrations.cloud_storage.azb.blob_endpoint_suffix
[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
|===
|Description
a|Azure blob storage endpoint suffix. You need to change this if you are not using Azure public cloud (e.g., if you are using Azure Government).
|Valid values
a|A string.
|Default value
m|+++blob.core.windows.net+++
|===

[role=label--enterprise-edition label--new-2025.03]
[[config_dbms.integrations.cloud_storage.azb.authority_endpoint]]
=== `dbms.integrations.cloud_storage.azb.authority_endpoint`

.dbms.integrations.cloud_storage.azb.authority_endpoint
[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
|===
|Description
a|Azure authority host endpoint (only required for certain methods of authentication, it should be specified in its full form - e.g., https://login.microsoftonline.com).
|Valid values
a|A string.
|Default value

> m|+++""+++

Author: ```suggestion
m|+++b''+++
```

> m|+++b''+++

Reviewer: this looks weird on the page

Author: I took from the test results. We're tied to the code here, and users will see the same when they run 'SHOW SETTINGS' cmd.

Author: https://live.neo4j-build.io/buildConfiguration/Documentation_Operations_Testing_ConfigurationSettingsPRs/23402203?showLog=23402203_6382_6173.6384&logFilter=debug&logView=flowAware

Reviewer: Th actual `AzbSettings` has a blank string so that `operations/test-config-settings.py:60` is doing something weird
```
    @Description(
            "Azure authority host endpoint (only required for certain methods of authentication, it should be specified in its full form - e.g., https://login.microsoftonline.com)")
    public static final Setting<String> authority_endpoint = newBuilder(
                    "dbms.integrations.cloud_storage.azb.authority_endpoint", STRING, "")
            .build();
```

Author: yep, you're right, the test is wrong 😄. The default has to be `" "`.  I tested this in Neo4j 2025.03

> m|+++""+++

Author: ```suggestion
m|++++++
```

> m|+++b''+++

Author: ```suggestion
m|+++""+++
```

|===

[role=label--enterprise-edition label--new-2025.03]
[[config_dbms.integrations.cloud_storage.gs.project_id]]
=== `dbms.integrations.cloud_storage.gs.project_id`

.dbms.integrations.cloud_storage.gs.project_id
[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
|===
|Description
a|Project ID of the Google storage bucket(s) to connect to, falling back to the value found by their SDK.
|Valid values

> a|

Author: ```suggestion
a|A string.
```

|Default value
m|++++++
|===

[role=label--enterprise-edition label--new-2025.03]
[[config_dbms.integrations.cloud_storage.s3.target_throughput_gbps]]
=== `dbms.integrations.cloud_storage.s3.target_throughput_gbps`

.dbms.integrations.cloud_storage.s3.target_throughput_gbps
[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
|===
|Description
a|The target throughput for transfer requests. Higher value means more connections will be established with S3. It's recommended to set it to the maximum network bandwidth on the host that the application is running on. The default is `10.0`, but when running on EC2 instances, this value can often be set much higher (being specific to the EC2 instance type).
|Valid values

> a|

Author: ```suggestion
a|An integer.
```

> a|An integer.

Author: ```suggestion
a|A double.
```

|Valid values
a|A comma-separated list where each element is A simple globbing pattern that can use `*` and `?`..
|Default value

> m|+++*bolt.connections*,*bolt.messages_received*,*bolt.messages_started*,*dbms.pool.bolt.free,*dbms.pool.bolt.total_size,*dbms.pool.bolt.total_used,*dbms.pool.bolt.used_heap,*cluster.raft.is_leader,*cluster.raft.last_leader_message,*cluster.raft.replication_attempt,*cluster.raft.replication_fail,*cluster.raft.last_applied,*cluster.raft.last_appended,*cluster.raft.append_index,*cluster.raft.commit_index,*cluster.raft.applied_index,*cluster.internal.discovery.memberset.left,*cluster.internal.discovery.crdt.gossip_id_data.size,*cluster.internal.discovery.crdt.server_data.size,*cluster.internal.discovery.crdt.database_data.size,*cluster.internal.discovery.crdt.leader_data.size,*cluster.internal.discovery.crdt.total_merge_operations,*cluster.internal.discovery.crdt.total_update_operations,*cluster.internal.discovery.gossip.incoming_queue_size,*cluster.internal.discovery.gossip.total_received_data,*cluster.internal.discovery.gossip.total_sent_data,*cluster.internal.discovery.gossip.uncontactable_members_exist,*check_point.*,*cypher.replan_events,*cypher.cache*,*ids_in_use*,*.neo4j.count.*,*pool.transaction.*.total_used,*pool.transaction.*.used_heap,*pool.transaction.*.used_native,*store.size*,*transaction.active_read,*transaction.active_write,*transaction.committed*,*transaction.last_committed_tx_id,*transaction.peak_concurrent,*transaction.rollbacks*,*page_cache.hit*,*page_cache.page_faults,*page_cache.usage_ratio,*vm.file.descriptors.count,*vm.gc.time.*,*vm.heap.used,*vm.memory.buffer.direct.used,*vm.memory.pool.g1_eden_space,*vm.memory.pool.g1_old_gen,*vm.pause_time,*vm.thread*,*db.query.execution*,*protocol*+++

Author: ```suggestion
m|+++*bolt.connections*,*bolt.messages_received*,*bolt.messages_started*,*dbms.pool.bolt.free,*dbms.pool.bolt.total_size,*dbms.pool.bolt.total_used,*dbms.pool.bolt.used_heap,*cluster.raft.is_leader,*cluster.raft.last_leader_message,*cluster.raft.replication_attempt,*cluster.raft.replication_fail,*cluster.raft.last_applied,*cluster.raft.last_appended,*cluster.raft.append_index,*cluster.raft.commit_index,*cluster.raft.applied_index,*cluster.internal.discovery.memberset.left,*cluster.internal.discovery.crdt.gossip_id_data.size,*cluster.internal.discovery.crdt.server_data.size,*cluster.internal.discovery.crdt.database_data.size,*cluster.internal.discovery.crdt.leader_data.size,*cluster.internal.discovery.crdt.total_merge_operations,*cluster.internal.discovery.crdt.total_update_operations,*cluster.internal.discovery.gossip.incoming_queue_size,*cluster.internal.discovery.gossip.total_received_data,*cluster.internal.discovery.gossip.total_sent_data,*cluster.internal.discovery.gossip.uncontactable_members_exist,*check_point.*,*cypher.replan_events,*cypher.cache*,*ids_in_use*,*.neo4j.count.*,*pool.transaction.*.total_used,*pool.transaction.*.used_heap,*pool.transaction.*.used_native,*store.size*,*transaction.active_read,*transaction.active_write,*transaction.committed*,*transaction.last_committed_tx_id,*transaction.peak_concurrent,*transaction.rollbacks*,*page_cache.hit*,*page_cache.page_faults,*page_cache.usage_ratio,*vm.file.descriptors.count,*vm.gc.time.*,*vm.heap.used,*vm.memory.buffer.direct.used,*vm.memory.pool.g1_eden_space,*vm.memory.pool.g1_old_gen,*vm.pause_time,*vm.thread*,*db.query.execution*,*protocol*+++ label:changed[Changed on 2025.03]
```
